frame/support/procedural/src/benchmark.rs

```diff
@@ -152,6 +173,41 @@ struct BenchmarkDef {
 impl BenchmarkDef {
 	/// Constructs a [`BenchmarkDef`] by traversing an existing [`ItemFn`] node.
 	pub fn from(item_fn: &ItemFn, extra: bool, skip_meta: bool) -> Result<BenchmarkDef> {
+		let invalid_return = |span| {
+			return Err(Error::new(
+				span,
+				"Invalid return type. Only `()` and `Result<(), BenchmarkError>` are allowed.",
```

Comment:
The `Result<(), BenchmarkError>` is actually a syntax thing and would still work if you alias `Result` to something else, or?

Reply:
yes, I agree an alias would be good to make this look better. I can use that assert impl macro I use elsewhere in this code to assert that whatever is put here is equivalent to `Result<(), BenchmarkError>` I believe